src/__tests__/calculate.test.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+  render, screen, fireEvent, cleanup,
+} from '@testing-library/react';
+import calculate from '../logic/calculate';
+import CalculatorPage from '../route/calculator';
+
+afterEach(() => cleanup());
+
+describe(('Testing the calculator functions'), () => {
+  it('test add operation', () => {
+    render(<CalculatorPage />);
+    const one = screen.getByText('5');
+    const sum = screen.getByText('+');
+    const two = screen.getByText('7');
+    const equals = screen.getByText('=');
+    const result = screen.getByTestId('result');
+    fireEvent.click(one);
+    fireEvent.click(sum);
+    fireEvent.click(two);
+    fireEvent.click(equals);
+    expect(result.textContent).toBe('12');
+  });
+
+  test('test subtract operation', () => {
+    render(<CalculatorPage />);
+    const one = screen.getByText('5');
+    const sub = screen.getByText('-');
+    const two = screen.getByText('7');
+    const equals = screen.getByText('=');
+    const result = screen.getByTestId('result');
+    fireEvent.click(one);
+    fireEvent.click(sub);
+    fireEvent.click(two);
+    fireEvent.click(equals);
+    expect(result.textContent).toBe('-2');
+  });
+
+  it('test multiply operation', () => {
```

Comment:
- [ ] Please always make sure that you are fulfilling the requirements before asking for the review, I tried to run the test initially when I ran all the tests at once. Then I was getting the failed `test` report but when I tried them specifically I came to know that your first three test cases are not passing and the problem is coming from `getByTestId('result');`. Kindly fix the issue and make sure that your test is passing before asking for another review.

### Problem 1 Screenshot

![Screenshot from 2023-08-31 14-10-36](https://github.com/misikir21/mama/assets/66476329/a7f9c5fe-3749-45bc-a9d2-adcb85346ab6)

### Problem 2 Screenshot

![Screenshot from 2023-08-31 14-08-04](https://github.com/misikir21/mama/assets/66476329/a5786db2-f10a-4c2f-99a9-79221ce0bfe5)